### PR TITLE
Fixed syndicate bombs being unpurchasable

### LINKED ISSUE #24400: Syndicate bomb not able to be purchased.
## Description
When you click on the syndicate bomb in the uplink nothing happens.

**Reproduction**
Tested as a admin converted syndicate and just with a normal debug uplink both did not work.
- open uplink find bomb post time limit.
- Cry when it doesn't purchase.

**Screenshots**
NA

**Additional context**
Little search in the recent PRs I see some modifications on #24022 and also searching back on the aHelp relays it lines up with when we started getting aHelps about it not working. @Ilya246 did you really test it? :trollface:

### PR DESCRIPTION
**Pull request by emmafornash**
## About the PR

Allows syndicate traitors and nukies to purchase Syndicate Bombs.

## Why / Balance

Resolves #24400

## Technical details

Adds a condition for Listings' `RestockTime` to the `Equals` method in `ListingPrototype.cs`.

## Media

https://github.com/space-wizards/space-station-14/assets/89596994/32b21261-375b-422b-a2ab-8b7ceb4d30ea

https://github.com/space-wizards/space-station-14/assets/89596994/1539874e-c7bc-41f4-aa86-6e2d1ec778cd

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Machines/bombs.yml [hunk 1]
```diff
@@ -127,6 +127,10 @@
       intensitySlope: 3
       maxIntensity: 400
 
+- type: entity
+  parent: SyndicateBomb
+  id: SyndicateBombNukeops
+
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Machines/bombs.yml:133]: What's the point of this?

Comment (emmafornash) [reply Resources/Prototypes/Entities/Structures/Machines/bombs.yml:133]: It's unclear from this code snippet, but in `uplink_catalog.yml`, the `productEntity` for the non-nukeops version of the listing is `SyndicateBomb`.

In the original code from #24022, shown below, the repeated `productEntity` of `SyndicateBomb` was causing the object to not be purchasable in either the traitor or nukeop uplink.
image

Creating a child of `SyndicateBomb` which had a different ID fixed this bug when used in place of the `productEntity` here.

However, I am new to this codebase, so if there's a better way to accomplish this, I would like to know.

Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Structures/Machines/bombs.yml:133]: This is just the entity being spawned, one has no bearing on the other.

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 2]
```diff
@@ -240,7 +240,7 @@
       - NukeOpsUplink
 
 - type: listing
-  id: UplinkSyndicateBombNukie
```
Comment (metalgearsloth) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:243]: We don't have prototype migrations, just mapping ones so this should be left as is.

Comment (emmafornash) [reply Resources/Prototypes/Catalog/uplink_catalog.yml]: Alright, thank you. I've reverted that change just now.

**Pull request merged**